### DISCUSSION TITLE
All possible observation text descriptions

### DISCUSSION THREAD

**Discussion by JohnENoonan** [weather-gov/api#547] [Q&A] [answered]
Hello all! I'm trying to find a list of the possible `textDescription`s of a current observation. For the request https://api.weather.gov/stations/KBWI/observations/latest I get 
``` js
"properties": {
         ...
        "textDescription": "Mostly Cloudy",
         ...
}
```

I need to process this description and it will be very useful to know what descriptions I need to account for. Is there a way to get all the possible strings this will return?

Thank you!

**Comment by scadergit** [comment, marked as answer]
The text description uses a phrase generator based on several factors that could change without notice to improve quality of service. We understand the need to reduce this down into a key that can be mapped, but that property is not available at this time. The best is to parse the properties for conditions that are relevant to your application. We plan to provide this feature in the future.

**Reply by JohnENoonan** [reply]
ok, thank you very much for the quick response!

**Reply by JohnENoonan** [reply]
tagging on to this question, If I'm going to parse the properties, if a result is Null am I to presume that means 0? For example
```
"precipitationLastHour": {
      "unitCode": "wmoUnit:m",
       "value": null,
       "qualityControl": "Z"
  },
```

**Reply by scadergit** [reply]
No, quality control of "Z" means preliminary, not QC'd, so null is likely not provided by the observation station. You can read more about NWS MADIS at https://madis.ncep.noaa.gov/madis_sfc_qc_notes.shtml (scroll down to QC Data Structures). MADIS quality control takes several passes at observations, performing quality control on the most important values first. Only temperture QC is required for the API to import the observation. Updates to the observation will be applied as received from MADIS.

**Comment by boehs** [comment]
For what it's worth, the deprecated icons api (https://api.weather.gov/icons) has a helpful list. All the APIs I want to use (observations, mostly), have a key like `"icon": "https://api.weather.gov/icons/land/day/skc?size=medium",` that's easy to parse out (`.match(/.*\/(.*)\?/)[1]`)

It's obvious why the icons API was deprecated, but what is the replacement? How do I know what icons to use based on

> a phrase generator based on several factors that could change without notice to improve quality of service

Is there anything? Do I keep using the icons API? (cc @scadergit)

Edit: See also

- https://github.com/weather-gov/api/discussions/557
- https://github.com/weather-gov/api/discussions/386
- https://github.com/weather-gov/api/discussions/273

**Comment by rex891** [comment]
Any status on whether this list is available, yet?
